### PR TITLE
[client-v2] Initial attempt at adding some server context to the client

### LINKED ISSUE #2027: [client-v2] Implement loading server context
### Description 
There is a need to load some server context after connecting to a server. Here is list of most important information:
- server timezone - it should be loaded to do proper DateTime conversion 
- user - it is important to know actual user if for example Bearer auth is used
- version - it is important to know when certain features are not working on older server versions and for debug purposes

### PR DESCRIPTION
**Pull request by Paultagoras**
## Summary
Closes #2027 
Adds a call to gather currentUser(), timezone(), and version() for the client

## Checklist
Delete items not relevant to your PR:
- [x] Closes issue 
- [x] Unit and integration tests covering the common scenarios were added
- [ ] A human-readable description of the changes was provided to include in CHANGELOG
- [ ] For significant changes, documentation in https://github.com/ClickHouse/clickhouse-docs was updated with further explanations or tutorials

### CODE REVIEW

File: client-v2/src/main/java/com/clickhouse/client/api/Client.java [hunk 1]
```diff
@@ -154,6 +154,10 @@ public class Client implements AutoCloseable {
 
     private final ColumnToMethodMatchingStrategy columnToMethodMatchingStrategy;
 
+    // Server context
+    private String serverUser;
+    private String serverVersion;
```
Comment (chernser) [inline client-v2/src/main/java/com/clickhouse/client/api/Client.java:159]: please add `getServerVersion()`

Comment (chernser) [reply client-v2/src/main/java/com/clickhouse/client/api/Client.java:159]: for the use we need to update `com.clickhouse.client.api.Client#getUser`

**Review by chernser — CHANGES_REQUESTED** [review]
please correct getters

**Review by chernser — APPROVED** [review]

**Review by mzitnik — COMMENTED** [review]
General comment i think we should a way to disable if the user wants

**Pull request merged**